File: erlang/p006.erl
-module(p006).

-export([start/0]).

f(N) ->
  L = lists:seq(1, N),
  SumSquares = lists:foldl(fun(X, Acc) -> Acc + X * X end, 0, L),
  SquareSum = lists:sum(L) * lists:sum(L),
  SquareSum - SumSquares.

start() ->
  io:format("~p ~n", [f(100)]).

File: c/p006.c
#include <stdio.h>
#include <stdbool.h>

int main(int argc, char** argv) {

    long sum_squares = 0;
    long sum = 0;

    for (long i = 1; i <= 100; ++i) {
        sum_squares += i * i;
        sum += i;
    }

    printf("%ld\n", (sum * sum - sum_squares));
}
